spec/forms/search_developer_form_spec.rb

```diff
@@ -3,16 +3,92 @@
+
+        @programming_language1 = create(:programming_language, name: 'ruby')
+        @programming_language2 = create(:programming_language, name: 'js')
+        @programming_language3 = create(:programming_language, name: 'c++')
+
+        @developer_language1 = create(:developer_language, developer_id: @developer1.id, language_id: @language1.id)
+        @developer_language2 = create(:developer_language, developer_id: @developer2.id, language_id: @language2.id)
+        @developer_language3 = create(:developer_language, developer_id: @developer3.id, language_id: @language3.id)
+        @developer_language4 = create(:developer_language, developer_id: @developer1.id, language_id: @language3.id)
+
+        @developer_programming1 = create(:developer_programming,
+                                         developer_id: @developer1.id,
+                                         programming_language_id: @programming_language1.id)
+        @developer_programming2 = create(:developer_programming,
+                                         developer_id: @developer2.id,
+                                         programming_language_id: @programming_language2.id)
+        @developer_programming3 = create(:developer_programming,
+                                         developer_id: @developer3.id,
+                                         programming_language_id: @programming_language3.id)
+        @developer_programming4 = create(:developer_programming,
+                                         developer_id: @developer1.id,
+                                         programming_language_id: @programming_language3.id)
+
       end
 
-      it 'returns no posts' do
-        form = SearchDeveloperForm.new({})
+      it 'return developer know a language' do
+        form = SearchDeveloperForm.new(developers_search_form: { language: @language1.code })
+        expect(form.search.size).to eq(1)
+      end
+
+      it 'return empty if nobody know a language' do
+        form = SearchDeveloperForm.new(developers_search_form: { language: 'cn' })
         expect(form.search.size).to eq(0)
       end
+
+      it 'return 2 developer know a language' do
+        form = SearchDeveloperForm.new(developers_search_form: { language: @language3.code })
+        expect(form.search.size).to eq(2)
+      end
+
+      it 'return 1 developer know a programming language' do
+        form = SearchDeveloperForm.new(developers_search_form: { programming_language: @programming_language1.name })
+        expect(form.search.size).to eq(1)
+      end
+
+      it 'return 2 developer know a programming language' do
+        form = SearchDeveloperForm.new(developers_search_form: { programming_language: @programming_language3.name })
+        expect(form.search.size).to eq(2)
+      end
+
+      it 'return empty if nobody know a programming language' do
+        form = SearchDeveloperForm.new(developers_search_form: { programming_language: 'kotlin' })
```

Comment:
Other developers besides you don't know if `kotlin` exists or not easily.
You should write `non_exisiting_programming_language` instead of `kotlin`, for instance, to tell your intention to others.